### ISSUE TITLE
Contributing.md: org wide or per-repo?

### ISSUE THREAD

**Issue by staltz** [ssb-js/ssb-js#29]
We put in a rule that every repo should have a CONTRIBUTING.md file, but so far none of the repos have this. So this is a good moment to ask ourselves: do we want the same ways-of-working (with git, with PR expectations, etc) everywhere, or do we allow it per-repo?

**Comment by christianbundy** [comment]
I'd love an org-wide one, and if repos have specific additions then they can add their own maybe?

If we create a repo at ssb-js/.github (example: https://github.com/github/.github) it'll automatically show up on each repo. (I hate that this is GitHub-specific, I'd way rather point to this repo being a source of truth.)

**Comment by staltz** [comment]
I guess a better question to kickstart this discussion is: what rules do we want to put in CONTRIBUTING.md? Arguably some rules that would go into CONTRIBUTING.md are already going into ssb-js/ssb-js/README.md.

If I'd make a CONTRIBUTING.md for my repos (`github.com/staltz/*`), I'd emphasize: (1) preferably open an issue before submitting a PR, (2) make small PRs because if they're rejected it doesn't hurt your feelings much, but if they're approved, you get a flow of good feelings getting every single one merged (i.e. microrewards) that overall make the contribution experience more fun, (3) breaking changes are reviewed with much more scrutiny than incremental improvements, prepare your expectations accordingly, (4) performance regressions are also taken seriously in reviews.

**Comment by staltz** [comment]
cc @mixmix

**Comment by mixmix** [comment]
I agree with all those things staltz named, except (1) - I've not personally felt the need for this, and if it's a small patch a clear description on the pr is perfect for me.

I haven't got into what is in contributing.md generally.
I imagine some broader context about what we value might be good. Like do we want any cool features? Or should you check a roadmap before making something big.
What help and sort can you expect if you're taking in an issue which but you're new to the code base?

And boundaries - what sort of contributions do we not want, what behavior do we expect? When will PRs be closed or issues locked?

I feel like we can also at detail as experience reveals need, so long as we all watch closely for pain points in contributing

Cc @arj03

**Comment by staltz** [comment]
> I agree with all those things staltz named, except (1) - I've not personally felt the need for this, and if it's a small patch a clear description on the pr is perfect for me.

Actually, I think I agree with you, I can easily drop (1) since in a world where we only have small PRs, I can't see the strict need to open issues beforehand.

> I imagine some broader context about what we value might be good. Like do we want any cool features? Or should you check a roadmap before making something big.

Yeah, it's good to start from big picture first. If you're literally asking me what I think, I believe repos in ssb-js should be conservative, they should be somewhat allergic to experimentation and big changes. Nothing against experimentation, of course, I just tend to see those belonging elsewhere (e.g. I still consider ssb-conn to be an experiment (although a maturing experiment), so I still host it under `https://github.com/staltz/*`, I wouldn't move it yet here)

>  What help and sort can you expect if you're taking in an issue which but you're new to the code base?

Good point

**Comment by arj03** [comment]
Agree with 2-4 also. Also lets try and get the ball rolling and add things as they come up.

As for stability, we might run into a situation where we need rather larger changes (like [new feed types](https://github.com/ssb-js/ssb-keys/pull/61) and in [ssb-ref](https://github.com/ssb-js/ssb-ref/pull/34)). But I think for situations like those it would be good to have a few people committing seeing then changes all the way through. It hurts looking over these major changes and seeing Christian doing them almost solo.

**Comment by mixmix** [comment]
I think this is ready for a simple PR?

**Comment by mixmix** [comment]
@arj03 i think you sugar a ghost point te large changes solo. I think as maintainers we need to practice knowing when to ask for help getting something through, and when to spot that help is needed (like you have here).

If /when we do a monthly call (when is that?) that would be a perfect time to do this check
